### PR TITLE
Supports export entity and  reference of ScriptComponent

### PR DESCRIPTION
**Pull request by JoneLau**

### CODE REVIEW

**Comment by JoneLau** [comment]
update cocos-creator/engine-3d#242

File: Assets/Editor/Exporter.Scene.cs [hunk 1]
```diff
@@ -359,7 +359,47 @@ public partial class Exporter {
...
+                      val = "child" + _nodes.IndexOf(prop.value.objectField as GameObject) + "@" + assetPath;
+                    } else {
+                      if (PrefabUtility.FindPrefabRoot(prop.value.objectField as GameObject) != prop.value.objectField) {
+                        Debug.LogWarning("Cannot refer to prefab child");
+                      }
+
+                      val = assetPath;
+                    }
+                  } else {
+                    val = assetPath;
```
Comment (jwu) [inline Assets/Editor/Exporter.Scene.cs]: val = Utils.AssetID(prop.value.objectField)

File: Assets/Editor/Exporter.Scene.cs [hunk 2]
```diff
@@ -359,8 +359,18 @@ public partial class Exporter {
...
+                // val = prop.value.objectField;
+                // if prefab reference another prefab,it will be null in asset
+                string assetPath = Utils.AssetID(prop.value.objectField);
+                if (!string.IsNullOrEmpty(assetPath)) {
+                  if (prop.value.objectField is GameObject) {
+                    if (PrefabUtility.FindPrefabRoot(prop.value.objectField as GameObject) != prop.value.objectField) {
+                      Debug.LogWarning("Cannot refer to prefab child");
+                    }
+                  }
+                  val = assetPath;
```
Comment (jwu) [inline Assets/Editor/Exporter.Scene.cs]: val = Utils.AssetID(prop.value.objectField);

File: Assets/Editor/Exporter.Scene.cs [hunk 3]
```diff
@@ -359,8 +359,18 @@ public partial class Exporter {
               } else if (item.type == PropType.String) {
                 val = prop.value.stringField;
               } else if (item.type == PropType.Reference) {
-                val = prop.value.objectField;
-              }
+                // val = prop.value.objectField;
+                // if prefab reference another prefab,it will be null in asset
+                string assetPath = Utils.AssetID(prop.value.objectField);
```
Comment (jwu) [inline Assets/Editor/Exporter.Scene.cs]: assetPath => assetID

File: Assets/Editor/Exporter.Scene.cs [hunk 4]
```diff
@@ -359,8 +359,18 @@ public partial class Exporter {
...
+                if (!string.IsNullOrEmpty(assetPath)) {
+                  if (prop.value.objectField is GameObject) {
```
Comment (jwu) [inline Assets/Editor/Exporter.Scene.cs]: 用  PrefabUtility.GetPrefabType() != PrefabType.None 去判断

File: Assets/Editor/Exporter.Scene.cs [hunk 5]
```diff
@@ -359,8 +359,18 @@ public partial class Exporter {
...
                 val = prop.value.stringField;
               } else if (item.type == PropType.Reference) {
-                val = prop.value.objectField;
-              }
+                // val = prop.value.objectField;
+                // if prefab reference another prefab,it will be null in asset
+                string assetID = Utils.AssetID(prop.value.objectField);
+                if (!string.IsNullOrEmpty(assetID)) {
+                  if (PrefabUtility.GetPrefabType(prop.value.objectField) !=PrefabType.None) {
+                    if (PrefabUtility.FindPrefabRoot(prop.value.objectField as GameObject) != prop.value.objectField) {
```
Comment (jwu) [inline Assets/Editor/Exporter.Scene.cs]: !=(空格)

**Pull request merged**